plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreConfig.java

```diff
@@ -240,6 +241,19 @@ public ThriftMetastoreConfig setTlsEnabled(boolean tlsEnabled)
         return this;
     }
 
+    public boolean isDisableHostNameVerification()
```

Comment:
Can we add some additional validation which ensure this config is enabled only if `tlsEnabled`

Reply:
Hey praveen, thank you for the comment.
I considered erroring out with a meaningful message if we enable this while ssl is disabled, but decided against it.

If you look at the other properties for ssl - they follow the same pattern - as in they only work if ssl is enabled, but don't error out if the `hive.metastore.thrift.client.ssl.enabled` isn't set/enabled.

This new property follows the same pattern.

Reply:
Hey praveen, thank you for the review.
I considered erroring out with a meaningful message if we enable this while ssl is disabled, but decided against it.

If you look at the other properties for ssl - they follow the following common pattern - as in they only work if ssl is enabled, but don't error out if the hive.metastore.thrift.client.ssl.enabled isn't set/enabled.

This new property follows the same pattern.

Please do lmk if you feel strongly otherwise and i'll make the edit.

Reply:
@Praveen2112 any update?

Reply:
Apologies for the late review. But we do apply `isTruststorePathValid` where we apply the validation. Plus disabling `hostname verification` doesn't makes sense in a non-ssl configuration. Other approach we could try is to extract the ssl-config to a separate class like in `MongoSslConfig` and bind the config if necessary